File: c++/types/integers/integer_size.cpp
#include <iostream>

int main()
{
  int i;
  signed int si; // also: signed
  unsigned int ui; // also: unsigned
  short int shi; // also: short
  long int loi; // also: long
  long long int loloi; // also: long long // from C++11
  signed short int sshi; // also: signed short
  signed long int sloi; // also: signed long
  signed long long int sloloi; // also: signed long long // from C++11
  unsigned short int ushi; // also: unsigned short
  unsigned long int uloi; // also: unsigned long
  unsigned long long int uloloi; // also: unsigned long long // from C++11

  std::cout << "int i; sizeof(i): " << sizeof(i) << std::endl
    << "sizeof(1): " << sizeof(1) << std::endl << std::endl;

  std::cout << "signed int si; sizeof(si): " << sizeof(si) << std::endl << std::endl;

  std::cout << "unsigned int ui; sizeof(ui): " << sizeof(ui) << std::endl
    << "sizeof(1u): " << sizeof(1u) << std::endl << std::endl;

  std::cout << "short int shi; sizeof(shi): " << sizeof(shi) << std::endl << std::endl;

  std::cout << "long int loi; sizeof(loi): " << sizeof(loi) << std::endl
    << "sizeof(1L): " << sizeof(1L) << std::endl << std::endl;

  std::cout << "long long int loloi; sizeof(loloi): " << sizeof(loloi) << std::endl
    << "sizeof(1LL): " << sizeof(1LL) << std::endl << std::endl;

  std::cout << "signed short int sshi; sizeof(sshi): " << sizeof(sshi) << std::endl << std::endl;

  std::cout << "signed long int sloi; sizeof(sloi): " << sizeof(sloi) << std::endl << std::endl;

  std::cout << "signed long long int sloloi; sizeof(sloloi): " << sizeof(sloloi) << std::endl << std::endl;

  std::cout << "unsigned short int ushi; sizeof(ushi): " << sizeof(ushi) << std::endl << std::endl;

  std::cout << "unsigned long int uloi; sizeof(uloi): " << sizeof(uloi) << std::endl
    << "sizeof(1uL): " << sizeof(1uL) << std::endl << std::endl;

  std::cout << "unsigned long long int uloloi; sizeof(uloloi): " << sizeof(uloloi) << std::endl
    << "sizeof(1uLL): " << sizeof(1uLL) << std::endl << std::endl;
}

File: c#/types/integers/IntegerSize.cs
using System;

class IntegerSize {
  static void Main(string[] args) {
    int i = 1;
    uint ui = 1;
    short shi = 1;
    long loi = 1;
    ushort ushi = 1;
    ulong uloi = 1;
    sbyte sy = 1;
    byte y = 1;

    System.Console.WriteLine(String.Format(
      "int i = 1; size of i: {0}\nsize of 1: {1}\n",
      System.Runtime.InteropServices.Marshal.SizeOf(i),
      System.Runtime.InteropServices.Marshal.SizeOf(1)
    ));

    System.Console.WriteLine(String.Format(
      "uint ui = 1; size of ui: {0}\nsize of 1u: {1}\n",
      System.Runtime.InteropServices.Marshal.SizeOf(ui),
      System.Runtime.InteropServices.Marshal.SizeOf(1u)
    ));

    System.Console.WriteLine(String.Format(
      "short shi; size of shi: {0}\n",
      System.Runtime.InteropServices.Marshal.SizeOf(shi)
    ));

    System.Console.WriteLine(String.Format(
      "long loi; size of loi: {0}\nsize of 1L: {1}\n",
      System.Runtime.InteropServices.Marshal.SizeOf(loi),
      System.Runtime.InteropServices.Marshal.SizeOf(1L)
    ));

    System.Console.WriteLine(String.Format(
      "ushort ushi; size of ushi: {0}\n",
      System.Runtime.InteropServices.Marshal.SizeOf(ushi)
    ));

    System.Console.WriteLine(String.Format(
      "ulong uloi; size of uloi: {0}\nsize of 1uL: {1}\n",
      System.Runtime.InteropServices.Marshal.SizeOf(uloi),
      System.Runtime.InteropServices.Marshal.SizeOf(1uL)
    ));

    System.Console.WriteLine(
      String.Format("byte sy = 1; size of sy: {0}\n",
      System.Runtime.InteropServices.Marshal.SizeOf(sy)
    ));

    System.Console.WriteLine(
      String.Format("byte y = 1; size of y: {0}\n",
      System.Runtime.InteropServices.Marshal.SizeOf(y)
    ));
  }
}
